tests/tracer/test_encoders.py

```diff
@@ -628,3 +640,35 @@ def run(self):
 
     unpacked = decode(encoder.encode(), reconstruct=True)
     assert unpacked is not None
+
+
+@pytest.mark.parametrize("encoder_cls", [JSONEncoder, JSONEncoderV2])
+def test_encode_traces_json_bytes(encoder_cls):
+    """
+    Regression test for: https://github.com/DataDog/dd-trace-py/issues/3115
+
+    Ensure we properly decode `bytes` objects when encoding with the JSONEncoder
+    """
+    # test encoding for JSON format
+    traces = [
+        [
+            Span(name=b"\x80span.a", tracer=None),
+            Span(name=u"\x80span.b", tracer=None),
+        ]
+    ]
```

Comment:
Since we set the encoding at the top of this file, one option here is to use a subprocess test for Python 2.7 to ensure we are testing under normal user conditions.